src/prog_models/models/uav_model/uav_model.py

```diff
@@ -0,0 +1,556 @@
+            }
+        # Check if passed final waypoint. If so, event_state is 0
+        if index_next > num_wypts:
+            return {
+                'TrajectoryComplete': 0
+            }
+        
+        t_next = self.parameters['waypoints']['waypoints_time'][index_next]
+        x_next = self.parameters['waypoints']['waypoints_x'][index_next]
+        y_next = self.parameters['waypoints']['waypoints_y'][index_next]
+        z_next = self.parameters['waypoints']['waypoints_z'][index_next]
+
+        # Define time interval for acceptable arrival at waypoint
+        time_buffer_left = (self.parameters['waypoints']['waypoints_time'][index_next] - self.parameters['waypoints']['waypoints_time'][index_next - 1])/2
+        if index_next == num_wypts:
+            # Final waypoint, add final buffer time 
+            time_buffer_right = t_next + self.parameters['final_time_buffer_sec']
+        else: 
+            time_buffer_right = (self.parameters['waypoints']['waypoints_time'][index_next+1] - self.parameters['waypoints']['waypoints_time'][index_next])/2
+
+        # Check if next waypoint is satisfied:
+        if x['t'] < t_next - time_buffer_left:
+            # Not yet within time of next waypoint
+            return {
+                    'TrajectoryComplete': (num_wypts - (index_next - 1))/num_wypts
+                }
+        elif t_next - time_buffer_left <= x['t'] <= t_next + time_buffer_right:
+            # Current time within ETA interval. Check if distance also within acceptable range
+            dist_now = np.sqrt((x['x']-x_next)**2 + (x['y']-y_next)**2 + (x['z']-z_next)**2)
+            if dist_now <= self.parameters['final_space_buffer_m']:
+                # Waypoint achieved
+                self.parameters['waypoints']['next_waypoint'] += 1
+                return {
+                    'TrajectoryComplete': (num_wypts - index_next)/num_wypts
+                }
+            else:
+                # Waypoint not yet achieved
+                return {
+                    'TrajectoryComplete': (num_wypts - (index_next - 1))/num_wypts
+                }
+        else:
+            # ETA passed before waypoint reached 
+            warn("Trajectory may not have reached waypoint associated with ETA of {})".format(t_next))
+            self.parameters['waypoints']['next_waypoint'] += 1
+            return {
+                    'TrajectoryComplete': (num_wypts - index_next)/num_wypts
+                }
+ 
+    def output(self, x : dict):
+        # Output is the same as the state vector
+        return self.OutputContainer({
+            'x': x['x'],
+            'y': x['y'],
+            'z': x['z'],
+            'phi': x['phi'],
+            'theta': x['theta'],
+            'psi': x['psi'],
+            'vx': x['vx'],
+            'vy': x['vy'],
+            'vz': x['vz'],
+            'p': x['p'],
+            'q': x['q'],
+            'r': x['r'],
+            't': x['t']
+            })
```

Comment:
```suggestion
        return self.OutputContainer(x.matrix)
```

Reply:
@teubert I found a mistake in my output function, since it includes time and we noted above that this isn't an output. I'll change this to "return self.OutputContainer(x.matrix[0:-1]) " . It seems to work - is that okay?

Reply:
sounds great